ansible/roles/ci/code_style_check/license/files/gather_missing_licences.py

```diff
@@ -78,10 +79,24 @@ def get_changes_in_pr(data):
         print(f"ERROR WITH COMMAND:\nstderr:{master_branch_process.stderr}\nstdout:{master_branch_process.stdout}")
         exit(1)
     devel_branches = ""
-    for branch in master_branch_process.stdout.split("\n"):
-        if any(entry in branch for entry in MASTER_BRANCHES):
-            devel_branches = branch.strip()
-            break
+    all_branches = master_branch_process.stdout.split("\n")
+    all_branches = [branch.strip() for branch in all_branches]
+    if  "noetic-devel" in all_branches:
+        devel_branches = "noetic-devel"
+    elif "melodic-devel" in all_branches:
+        devel_branches = "melodic-devel"
+    elif "kinetic-devel" in all_branches:
+        devel_branches = "kinetic-devel"
+    elif "jade-devel" in all_branches:
+        devel_branches = "jade-devel"
+    elif "indigo-devel" in all_branches:
+        devel_branches = "indigo-devel"
+    elif "devel" in all_branches:
+        devel_branches = "devel"
+    elif "master" in all_branches:
+        devel_branches = "master"
+    elif "main" in all_branches:
+        devel_branches = "main"
     if devel_branches == "":
         print(f"Could not find the master branch: checks for {MASTER_BRANCHES}")
```

Comment:
you're checking for more than only MASTER_BRANCHES, include also the ros_distro-devel branches

Reply:
Its not checking pervious versions, its finding the first thing that matches one of these repo names. So it will look for noetic, then melodic ... then if it doesnt find them then it just looks for either devel, main or master. This covers all the names of our master branches.